frontend/web/components/release-pipelines/SinglePipelineStageAction.tsx

```diff
@@ -24,42 +25,65 @@ const getSegment = (action: StageActionRequest | undefined) => {
 }
 
 interface SinglePipelineStageActionProps {
+  indexPos?: number
   action?: StageActionRequest
-  onActionChange: (option: { value: string; label: string }) => void
-  onSegmentChange: (option: { value: number; label: string }) => void
+  headerText?: string
+  onRemoveAction?: (id: number) => void
+  onActionChange: (option: { value: string; label: string }, id: number) => void
+  onSegmentChange: (
+    option: { value: number; label: string },
+    id: number,
+  ) => void
   isSegmentsLoading: boolean
   segmentOptions: { label: string; value: number }[] | undefined
 }
 
 const SinglePipelineStageAction = ({
   action,
+  headerText = 'Then',
+  indexPos = 0,
   isSegmentsLoading,
   onActionChange,
+  onRemoveAction,
   onSegmentChange,
   segmentOptions,
 }: SinglePipelineStageActionProps) => {
   return (
     <>
       <Row>
         <div className='flex-1 and-divider__line' />
-        <div className='mx-2'>Then</div>
+        <div className='mx-2'>{headerText}</div>
```

Comment:
Maybe we can avoid this prop and have this directly as a variable in this component wdyt ?
`headerText={index === 0 ? 'Then' : 'And'}`